scripts/pre_commit_linter.py

```diff
@@ -714,11 +718,144 @@ def _check_spacing(all_files):
     return summary_messages
 
 
+def _check_comments(all_files):
+    """This functions ensures that comments end in a period."""
+    print 'Starting comment checks'
+    print '----------------------------------------'
+    summary_messages = []
+    files_to_check = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)
+        and filename.endswith('.py')]
+    message = 'There should be a period at the end of the comment.'
+    failed = False
+    for filename in files_to_check:
+        f = open(filename, 'r')
+        file_content = f.readlines()
+        file_length = len(file_content)
+        for line_num in range(file_length):
+            line = file_content[line_num].lstrip().rstrip()
+            next_line = ""
+            if line_num + 1 < file_length:
+                next_line = file_content[line_num + 1].lstrip().rstrip()
+
+            if line.startswith('#') and not next_line.startswith('#'):
+                # Check for '.' or '?' at end of comment.
+                if (
+                    line[-1] not in ALLOWED_TERMINATING_PUNCTUATIONS
+                    and not any (word in line
```

Comment:
Move the `and` to the previous line. Move the statement below to this line. See [this snippet](https://github.com/oppia/oppia/blob/c6df032ba9e3671b8401aa4082ff62edb2e2cc05/scripts/deploy.py#L214) for an example.